src/main/scala/DataMunging.scala

```diff
@@ -0,0 +1,40 @@
+package DataMunging
+import scala.io.Source
+
+object DataMunging {
+
+    def main(args: Array[String]) ={
+            getMinSpread()
+            getMinGoalDiff()
+    }
+    def getMinSpread() = {
+        val source = Source.fromFile("weather.dat")
+        val spread = source.getLines.toList
+                               .drop(2) //the first line is a header and second line is a blank space so skip those lines here.
+                               .map(_.split("\\s+")) // split on regex white space
+                               .map(x => (x(1), x(2).stripSuffix("*").toDouble-x(3).stripSuffix("*").toDouble)) //clean the recourds as somelines have * and aslo conver to double
+        val min_spread = find_min(spread) // get minimum spread
+        println(s"1: the day with the smallest temparature spread is day ${min_spread._1} with a spread of ${min_spread._2} degrees")
```

Comment:
println at this point is okay. However your getMinSpread() function can be more testable with a unit test if you make it return the actual spread. In other words, making it 'a pure function without side-effects' . 
The println can be cone in the main method. So your main method would look like:

```
def main(args: Array[String]) ={
    val minSpread = getLines("weather.dat"). getMinSpread
    println(s"1: the day with the smallest temparature spread is day ${minSpread._1} with a spread of ${minSpread._2} degrees")
    val team = getLines("football.dat").getMinGoalDiff()
    println(s"2: the team with the smallest difference for and against goal is ${team._1} with a difference of ${team._2} degrees")
 }
```